File: frontend/src/services/supplierContract.ts
/* 🅵-1: supplierContract.ts - kill the `any` */
import { api } from './api';

export interface SupplierContractRecord {
  id: number;
  supplier_id: number;
  contract_no: string;
  sign_date: string | null;
  start_date: string | null;
  end_date: string | null;
  amount: number | null;
  status: string | null;
  remark: string | null;
  created_at: string;
  updated_at: string;
  is_deleted: boolean;
}

export interface SupplierContractCreatePayload {
  supplier_id: number;
  contract_no: string;
  sign_date?: string | null;
  start_date?: string | null;
  end_date?: string | null;
  amount?: number | null;
  status?: string | null;
  remark?: string | null;
}

export const fetchSupplierContracts = (params?: { supplier_id?: number; page?: number; page_size?: number }) => {
  const q = new URLSearchParams();
  q.set('page', String(params?.page || 1));
  q.set('page_size', String(params?.page_size || 20));
  if (params?.supplier_id) q.set('supplier_id', String(params.supplier_id));
  return api.get<{ items: SupplierContractRecord[]; total: number; page: number; page_size: number }>('/supplier-contracts?' + q.toString());
};

export const createSupplierContract = (data: SupplierContractCreatePayload) =>
  api.post<SupplierContractRecord>('/supplier-contracts', data);

export const updateSupplierContract = (id: number, data: Partial<SupplierContractCreatePayload>) =>
  api.patch<SupplierContractRecord>(`/supplier-contracts/${id}`, data);

export const deleteSupplierContract = (id: number) =>
  api.delete<void>(`/supplier-contracts/${id}`);

File: backend/app/schemas/supplier_contract.py
"""供应商合同 Schema。"""
from __future__ import annotations
from datetime import date, datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict, Field
from app.schemas.base import FinanceDeskBaseModel


class SupplierContractCreate(FinanceDeskBaseModel):
    supplier_id: int = Field(..., description="供应商ID")
    contract_no: str = Field(..., max_length=100, description="合同编号")
    sign_date: Optional[date] = Field(None, description="签订日期")
    start_date: Optional[date] = Field(None, description="合同开始日期")
    end_date: Optional[date] = Field(None, description="合同结束日期")
    amount: Optional[float] = Field(None, description="合同金额")
    status: Optional[str] = Field("有效", max_length=50, description="合同状态")
    remark: Optional[str] = Field(None, description="备注")


class SupplierContractUpdate(FinanceDeskBaseModel):
    contract_no: Optional[str] = Field(None, max_length=100)
    sign_date: Optional[date] = None
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    amount: Optional[float] = None
    status: Optional[str] = Field(None, max_length=50)
    remark: Optional[str] = None


class SupplierContractResponse(FinanceDeskBaseModel):
    id: int
    supplier_id: int
    contract_no: str
    sign_date: Optional[date] = None
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    amount: Optional[float] = None
    status: Optional[str] = None
    remark: Optional[str] = None
    created_at: datetime
    updated_at: datetime
    is_deleted: bool


class PaginatedSupplierContracts(FinanceDeskBaseModel):
    items: list[SupplierContractResponse]
    total: int
    page: int
    page_size: int
